ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/PrivateStateRootResolverTest.java

```diff
@@ -0,0 +1,180 @@
+    final PrivateStateRootResolver privateStateRootResolver =
+        new PrivateStateRootResolver(privateStateStorage);
+    assertThat(
+            privateStateRootResolver.resolveLastStateRoot(
+                privacyGroupId, BLOCKCHAIN.getChainHeadHeader().getHash()))
+        .isEqualTo(PrivateStateRootResolver.EMPTY_ROOT_HASH);
+  }
+
+  @Test
+  public void ifCommitmentForPrivacyGroupExistsReturnsRootHash() {
+    final PrivateStateStorage.Updater updater = privateStateStorage.updater();
+    updater.putPrivateBlockMetadata(
+        BLOCKCHAIN.getBlockByNumber(16).get().getHash(),
+        Bytes32.wrap(privacyGroupId),
+        new PrivateBlockMetadata(
+            Collections.singletonList(
+                new PrivateTransactionMetadata(
+                    BLOCK_GENERATOR.transaction().getHash(), pmt1StateHash))));
+    updater.putPrivacyGroupHeadBlockMap(
+        BLOCKCHAIN.getChainHeadHash(),
+        new PrivacyGroupHeadBlockMap(
+            Collections.singletonMap(
+                Bytes32.wrap(privacyGroupId), BLOCKCHAIN.getBlockByNumber(16).get().getHash())));
+    updater.commit();
+    final PrivateStateRootResolver privateStateRootResolver =
+        new PrivateStateRootResolver(privateStateStorage);
+    assertThat(
+            privateStateRootResolver.resolveLastStateRoot(
+                privacyGroupId, BLOCKCHAIN.getChainHeadHash()))
+        .isEqualTo(pmt1StateHash);
+  }
+
+  @Test
+  public void ifMultipleCommitmentsExistsReturnsRootHashForCorrectPrivacyGroup() {
+    final PrivateStateStorage.Updater updater = privateStateStorage.updater();
+    updater.putPrivateBlockMetadata(
+        BLOCKCHAIN.getBlockByNumber(16).get().getHash(),
+        Bytes32.wrap(privacyGroupId),
+        new PrivateBlockMetadata(
+            Collections.singletonList(
+                new PrivateTransactionMetadata(
+                    BLOCK_GENERATOR.transaction().getHash(), pmt1StateHash))));
+    updater.putPrivateBlockMetadata(
+        BLOCKCHAIN.getBlockByNumber(16).get().getHash(),
+        Bytes32.wrap(failingPrivacyGroupId),
+        new PrivateBlockMetadata(
+            Collections.singletonList(
+                new PrivateTransactionMetadata(
+                    BLOCK_GENERATOR.transaction().getHash(), pmt2StateHash))));
+    updater.putPrivacyGroupHeadBlockMap(
+        BLOCKCHAIN.getChainHeadHash(),
+        new PrivacyGroupHeadBlockMap(
+            Collections.singletonMap(
+                Bytes32.wrap(privacyGroupId), BLOCKCHAIN.getBlockByNumber(16).get().getHash())));
+    updater.commit();
+    final PrivateStateRootResolver privateStateRootResolver =
+        new PrivateStateRootResolver(privateStateStorage);
+    assertThat(
+            privateStateRootResolver.resolveLastStateRoot(
+                privacyGroupId, BLOCKCHAIN.getChainHeadHash()))
+        .isEqualTo(pmt1StateHash);
+  }
+
+  @Test
+  public void ifMultipleCommitmentsForSamePrivacyGroupExistsReturnsLatestStateRoot() {
```

Comment:
NIT: `resolveLatestRootHashWhenMultipleCommitmentsForTheSamePrivacyGroupExist` would match the most common naming pattern used for unit tests in Besu.